qa/L0_decoupled/decoupled_test.py

```diff
@@ -277,6 +290,42 @@ def test_no_streaming(self):
         self._no_streaming_helper("grpc")
         self._no_streaming_helper("http")
 
+    def test_wrong_shape(self):
+        # Sends mismatching shapes for IN and DELAY. Server should return
+        # appropriate error message. The shape of IN is [repeat_count],
+        # where as shape of DELAY is [repeat_count + 1].
+
+        data_offset = 100
+        repeat_count = 1
+        delay_time = 1000
+        wait_time = 2000
+
+        input_data = np.arange(start=data_offset,
+                               stop=data_offset + repeat_count,
+                               dtype=np.int32)
+        delay_data = (np.ones([repeat_count + 1], dtype=np.uint32)) * delay_time
+        wait_data = np.array([wait_time], dtype=np.uint32)
+
+        # Initialize data for IN
+        self.inputs_[0].set_shape([repeat_count])
+        self.inputs_[0].set_data_from_numpy(input_data)
+
+        # Initialize data for DELAY
+        self.inputs_[1].set_shape([repeat_count + 1])
+        self.inputs_[1].set_data_from_numpy(delay_data)
+
+        # Initialize data for WAIT
+        self.inputs_[2].set_data_from_numpy(wait_data)
+
+        user_data = UserData()
+        result_dict = {}
+
+        try: 
```

Comment:
The unittest library has `assertRaises()` for enforcing exceptions:
```
with self.assertRaises(InferenceServerException) as cm:
    self._stream_infer(1, repeat_count, user_data, result_dict)

self.assertIn("expected IN and DELAY shape to match, got [1] and [2]", str(cm.exception))
``` 

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises